std/math.d

```diff
@@ -1585,10 +1592,21 @@ real exp(real x) @trusted pure nothrow @nogc
         enum real C2 = 1.428606820309417232121458176568075500134E-6L;
 
         // Overflow and Underflow limits.
-        enum real OF =  1.1356523406294143949492E4L;
-        enum real UF = -1.1432769596155737933527E4L;
+        static if (real.mant_dig == 64) // 80-bit reals
+        {
+            enum real OF =  1.1356523406294143949492E4L; // ln((1-2^-64) * 2^16384) ≈ 0x1.62e42fefa39efp+13
+            enum real UF = -1.1398805384308300613366E4L; // ln(2^-16445) ≈ -0x1.6436716d5406ep+13
+        }
+        else static if (real.mant_dig == 53) // 64-bit reals
+        {
+            enum real OF =  0x1.62e42fefa39efp+9L; // ln((1-2^-53) * 2^1024) = 709.78271...
+            enum real UF = -0x1.74385446d71c3p+9L; // ln(2^-1074) = -744.44007...
```

Comment:
Likewise, the smallest denormal number is ``ln(2^-1075)`` - I'll raise a PR to correct these.